### ISSUE TITLE
Functional constructor syntax for XML item types

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/ballerina-spec#224]
Putting #176 and #2 together, we should allow XML item types to be functional constructors.

- `xml(x1, x2, x3)` creates an XML sequence by concatenating x1, x2,...
- `xmlcomment(str)` creates an XML comment
- `xmlpi(target, str)` creates an XML processing instruction
- `xmlelement(tag, att1=val1, att2=val2)` creates an XML element

Note the parallel between xmlelement and error:
- error is typed as `error<StringType,RecordType>`
- xmlelement will be typed as `xmlelement<StringType,RecordType,ChildrenType>` (see #177)
- you can specify the fields of the error detail record as named arguments in the error functional constructor
- you can specify the fields of the attribute map as named arguments in the xmlelement functional constructor

This is really powerful because it allows you to use binding patterns to destructure XML elements and other XML items (particularly in conjunction with #107).

But to really make this work well we need a way to specify the children of the XML element in the constructor. Just using a positional argument is problematic:

```
type P xmlelement<"p", record { string id?; }>;
var x = P("foo");
```
Does "foo" override the contextually implied string name from the P type or is it part of the content? This may influence the design of #223.

**Comment by jclark** [comment]
Given how we have done #223, the named XML element constructor would work like this:
```
type P xmlelement<"p", record { string id?; }>;
var x = P(id = "p1");
```

**Comment by jclark** [comment]
I think it will work out better if the children are syntactically distinguished from the other arguments both in constructors (now) and precise types (later, see #251), e.g.

```
var x = xmlelement("p"; "This is", xmlelement("b"; "bold"));
type P xmlelement<"p"; (xmlchar|InlineElement)*>;
```

**Comment by jclark** [comment]
XML functional constructors should be constants. But what about error constructors?

**Comment by jclark** [comment]
This was fixed by 7ffe36128182add34d1e2fcd82514adf5acc9c1b
